src/functions.php

```diff
@@ -75,8 +108,27 @@ function debug_resource($value = null)
  * @throws \RuntimeException if no viable Handler is available.
  *
  * @return callable Returns the best handler for the given system.
+ *
+ * @deprecated
  */
 function choose_handler(): callable
+{
+    @trigger_error(sprintf('Function %s is deprecated and will be removed on guzzle 8.0', __FUNCTION__), E_USER_DEPRECATED);
```

Comment:
I'm not sure about this one, because:
1. it's already in use in [_public_ documentation](https://github.com/guzzle/guzzle/blob/e889719ced2eefd498844550d69a99e681ea4c43/docs/handlers-and-middleware.rst)
2. I definitely see a use case described above, when you want to create an empty `HandlerStack` with the default handler, IMO it's legit.